## HDCS Fall 22 internship project #1: Stage A Graded Quiz
INTRODUCTION TO PYTHON FOR MACHINE LEARNING
### Import the required libraries


In [73]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

In [72]:
data = pd.read_csv("Food_Africa.csv", encoding = "latin-1", on_bad_lines='skip')

/usr/local/lib/python3.7/dist-packages/IPython/core/interactiveshell.py:3326: DtypeWarning: Columns (2) have mixed types.Specify dtype option on import or set low_memory=False.
  exec(code_obj, self.user_global_ns, self.user_ns)


In [74]:
data.head()

Area Code     Area Item Code         Item  Element Code  \
0         4  Algeria      2501   Population         511.0   
1         4  Algeria      2501   Population        5301.0   
2         4  Algeria      2901  Grand Total         664.0   
3         4  Algeria      2901  Grand Total         674.0   
4         4  Algeria      2901  Grand Total         684.0   

                                  Element             Unit     Y2014  \
0           Total Population - Both sexes     1000 persons  38924.00   
1                Domestic supply quantity      1000 tonnes      0.00   
2           Food supply (kcal/capita/day)  kcal/capita/day   3377.00   
3  Protein supply quantity (g/capita/day)     g/capita/day     94.90   
4      Fat supply quantity (g/capita/day)     g/capita/day     80.06   

      Y2015     Y2016     Y2017     Y2018  
0  39728.00  40551.00  41389.00  42228.00  
1      0.00      0.00      0.00      0.00  
2   3379.00   3372.00   3341.00   3322.00  
3     94.35     94.72     92.82     91.83  
4     79.36     77.40     80.19     77.28

### Total Protein supply quantity in Madagascar in 2015
The answer is 346.10 as we can see below

In [76]:
Madagascar_data = data[data["Area"] == "Madagascar"]
Madagascar_data.groupby("Element")["Y2015"].sum()

Element
Domestic supply quantity                  62429.96
Export Quantity                             989.50
Fat supply quantity (g/capita/day)          183.70
Feed                                       4140.44
Food                                      42241.30
Food supply (kcal/capita/day)             15370.00
Food supply quantity (kg/capita/yr)        1743.18
Import Quantity                            3443.60
Losses                                     4352.00
Other uses (non-food)                      7914.24
Processing                                 3606.00
Production                                58965.78
Protein supply quantity (g/capita/day)      346.10
Residuals                                 -1232.00
Seed                                       1398.00
Stock Variation                           -1010.00
Total Population - Both sexes             48468.00
Name: Y2015, dtype: float64

### The total number of the sum of Processing in 2017
The answer is 475887.00 as we see

In [77]:
data.groupby("Element")["Y2017"].sum()

Element
3.000000                                        0.00
4.000000                                        0.00
Domestic supply quantity                  3450498.02
Export Quantity                            328827.00
Fat supply quantity (g/capita/day)          16647.68
Feed                                       380969.42
Food                                      2069626.16
Food supply (kcal/capita/day)              735156.00
Food supply quantity (kg/capita/yr)         78710.07
Import Quantity                            452426.37
Losses                                     266646.00
Other uses (non-food)                      150502.13
Processing                                 475887.00
Production                                3426133.58
Protein supply quantity (g/capita/day)      19096.36
Residuals                                   69376.00
Seed                                        37283.28
Stock Variation                             99609.06
Total Population - Both sexes         

### The total of Wine produced in 2015 and 2018
The answer is 2409.00 for 2015 & 2121.00 for 2018

In [80]:
Wine_data = data[data['Item'] == "Wine"]
Wine_data.groupby("Element")["Y2015"].sum()

Element
Domestic supply quantity                  1702.00
Export Quantity                           1020.00
Fat supply quantity (g/capita/day)           0.00
Food                                      1366.00
Food supply (kcal/capita/day)              234.00
Food supply quantity (kg/capita/yr)        126.04
Import Quantity                            513.00
Losses                                       2.00
Other uses (non-food)                       20.00
Processing                                 315.00
Production                                2409.00
Protein supply quantity (g/capita/day)       0.00
Residuals                                    0.00
Stock Variation                            194.00
Tourist consumption                          0.00
Name: Y2015, dtype: float64

In [81]:
Wine_data.groupby("Element")["Y2018"].sum()

Element
Domestic supply quantity                  1749.0
Export Quantity                           1105.0
Fat supply quantity (g/capita/day)           0.0
Food                                      1410.0
Food supply (kcal/capita/day)              241.0
Food supply quantity (kg/capita/yr)        128.9
Import Quantity                            550.0
Losses                                       4.0
Other uses (non-food)                       58.0
Processing                                 280.0
Production                                2121.0
Protein supply quantity (g/capita/day)       0.0
Residuals                                    0.0
Stock Variation                           -185.0
Tourist consumption                          0.0
Name: Y2018, dtype: float64

### The Area with the highest sum in 2017
The answer is Nigeria with sum of 2966536.46

In [84]:
data.groupby("Area")["Y2017"].sum().sort_values(ascending = False)

Area
Nigeria                        2966536.46
South Africa                   1035181.08
Egypt                           866379.92
Morocco                         776990.72
Ghana                           675198.12
United Republic of Tanzania     645233.70
Kenya                           529321.32
Sudan                           479863.84
Ethiopia                        448683.76
Uganda                          427900.76
Malawi                          362197.42
Algeria                         325644.27
Mozambique                      322815.96
Mauritania                      313330.92
Mali                            299856.66
Madagascar                      262395.46
Niger                           253415.16
Tunisia                         248334.40
Cameroon                        232030.43
Angola                          229159.57
Côte d'Ivoire                   224599.01
Zambia                          206447.54
Guinea                          196277.74
Senegal                      

### The total number of unique countries in the dataset

In [85]:
 data.Area.nunique()

50

### The mean and standard deviation across the whole dataset for the year 2017

In [90]:
print(f"The mean is: " + str(data.Y2017.mean()) + ", and the standar deviation is: " + str(data.Y2017.std())) 

The mean is: 143.17531791907513, and the standar deviation is: 1730.3772175725207


### The Areas with the 7th lowest sum in 2017
The answer is Cabo Verde with 14650.74



In [91]:
data.groupby("Area")["Y2017"].sum().sort_values(ascending = True)


Area
kcal/capita/day                      0.00
Sudan (former)                       0.00
Ethiopia PDR                         0.00
Comoros                             59.84
Seychelles                         884.68
Cabo Verde                       14650.74
Botswana                         22101.30
Djibouti                         22729.91
Sao Tome and Principe            25325.26
Gabon                            27979.64
Gambia                           28153.48
Central African Republic         29937.00
Guinea-Bissau                    38205.54
Congo                            41181.68
Lesotho                          42535.92
Eswatini                         54343.33
Liberia                          58684.40
Namibia                          59749.78
Chad                             71594.68
Togo                             99683.76
Burkina Faso                    101855.07
Mauritius                       102221.66
Sierra Leone                    110622.66
Benin                        

### The year with the highest sum of Stock Variation?
The answer is 2017 with a Stock Variation of 99609.06

In [93]:
data.groupby("Element")[["Y2014","Y2015","Y2016","Y2017"]].sum()

Y2014       Y2015       Y2016  \
Element                                                                      
3.000000                                      4.00        4.00        0.00   
4.000000                                      0.00        0.00        0.00   
Domestic supply quantity                3313110.86  3337541.62  3376385.64   
Export Quantity                          271005.89   286742.13   273387.01   
Fat supply quantity (g/capita/day)        16594.28    16600.33    16396.21   
Feed                                     370315.45   382097.74   389283.37   
Food                                    1996964.41  2028363.73  2053280.71   
Food supply (kcal/capita/day)            734680.00   732173.00   729863.00   
Food supply quantity (kg/capita/yr)       80088.16    79618.72    79064.99   
Import Quantity                          416863.54   400868.36   436504.34   
Losses                                   255691.00   258006.00   261818.00   
Other uses (non-food)                    128802.31   102801.54   109024.19   
Processing                               468224.00   471675.00   455532.00   
Production                              3257168.63  3277685.07  3262242.24   
Protein supply quantity (g/capita/day)    19051.97    19021.22    18964.17   
Residuals                                 59157.00    58087.00    72449.00   
Seed                                      33325.84    35957.64    34576.40   
Stock Variation                           90053.68    54210.96    48620.00   
Total Population - Both sexes           1704208.00  1747903.00  1792531.00   
Tourist consumption                         469.00      407.00      148.00   

                                             Y2017  
Element                                             
3.000000                                      0.00  
4.000000                                      0.00  
Domestic supply quantity                3450498.02  
Export Quantity                          328827.00  
Fat supply quantity (g/capita/day)        16647.68  
Feed                                     380969.42  
Food                                    2069626.16  
Food supply (kcal/capita/day)            735156.00  
Food supply quantity (kg/capita/yr)       78710.07  
Import Quantity                          452426.37  
Losses                                   266646.00  
Other uses (non-food)                    150502.13  
Processing                               475887.00  
Production                              3426133.58  
Protein supply quantity (g/capita/day)    19096.36  
Residuals                                 69376.00  
Seed                                      37283.28  
Stock Variation                           99609.06  
Total Population - Both sexes           1838048.00  
Tourist consumption                         152.00

### The year with the least correlation with ‘Element Code’
The answer is 2016, it has 2.33% correlation with ‘Element Code’


In [94]:
data.corr()

Element Code     Y2014     Y2015     Y2016     Y2017     Y2018
Element Code      1.000000  0.024503  0.023838  0.023304  0.024268  0.024321
Y2014             0.024503  1.000000  0.994156  0.995830  0.995128  0.995136
Y2015             0.023838  0.994156  1.000000  0.995233  0.986640  0.987140
Y2016             0.023304  0.995830  0.995233  1.000000  0.991948  0.992262
Y2017             0.024268  0.995128  0.986640  0.991948  1.000000  0.998111
Y2018             0.024321  0.995136  0.987140  0.992262  0.998111  1.000000

### The total number and percentage of missing data in 2014

In [95]:
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 99504 entries, 0 to 99503
Data columns (total 12 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   Area Code     99504 non-null  object 
 1   Area          99502 non-null  object 
 2   Item Code     99503 non-null  object 
 3   Item          99503 non-null  object 
 4   Element Code  99502 non-null  float64
 5   Element       99502 non-null  object 
 6   Unit          99502 non-null  object 
 7   Y2014         96880 non-null  float64
 8   Y2015         96953 non-null  float64
 9   Y2016         97005 non-null  float64
 10  Y2017         97053 non-null  float64
 11  Y2018         97213 non-null  float64
dtypes: float64(6), object(6)
memory usage: 9.1+ MB


In [100]:
print("The total numbrt of missing values in 2014 is: " + str(99504-96880))

The total numbrt of missing values in 2014 is: 2624


In [98]:
print("The percentage of missing values in 2014 is: " + str(100-(96880/99504)*100) +"%")

The percentage of missing values in 2014 is: 2.637079916385261%
